geode-core/src/test/java/org/apache/geode/test/junit/rules/MemberStarterRule.java

```diff
@@ -105,6 +105,11 @@ public void after() {
     }
   }
 
+  public T withPort(int memberPort) {
```

Comment:
do we need this? we avoid using a specific port for those tests. We use RandomAvailablePortHelper to determine an available port at runtime,

Reply:
You're right, we don't need this for this PR anymore. But there does not seem to be another way to specify which port that you want to use, so should we keep it anyway?

Reply:
How do you know if the port is available on a given system?  If the port is in use the test will fail.

Reply:
There is a method that will give you an available port. See the [DUnit test](https://github.com/apache/geode/pull/2066/files/454bd87f5cd83d1d257164092978e29e84aca3c7#diff-a728b1a4e3025151233500629689d1ecR50) for this method

Reply:
But this does not provide any insurance that the locator won't fail to start on that port. I'm fine with removing this if this implementation is insufficient, I just thought it would be nice to keep if it is good as is.

Reply:
the purpose of memberStarterRule or ClusterStarterRule is that you don't need to explicitly use AvailablePort.getRandomAvailablePort to get a random available port. This rule should handle all that for you.

Reply:
actually we just found a use case that this withPort is useful. After we crashed a server, we want to bring the server back up on the same port.